sql/proposals/proposals.go

```diff
@@ -4,11 +4,31 @@ import (
 	"fmt"
 	"io"
 
+	"github.com/spacemeshos/go-scale"
+
 	"github.com/spacemeshos/go-spacemesh/codec"
 	"github.com/spacemeshos/go-spacemesh/common/types"
 	"github.com/spacemeshos/go-spacemesh/sql"
 )
 
+// TransactionIDs is a list of transaction IDs.
+type TransactionIDs []types.TransactionID
```

Comment:
shouldn't these belong in common/types?

Reply:
good question. This type is specifically for sql/proposals with the only place where we encode/decode `[]types.TransactionID` and need this special type with proper `EncodeScale/DecodeScale` on it. So not sure. @dshulyak what do you think?

Reply:
one would assume that there are similar issues with ballots's votes where a list of BlockIDs are encoded, sometimes empty. can we have consistent solutions with that too?

Reply:
@countvonzero I can't find where we encode a list of BlockIDs as a slice. We have it as a list of structs but then it's not an issue and doesn't need this workaround.

Reply:
i don't understand why is it in sql module. i also not aware where we need to encode BlockIDs separately

lets use scale.Encode/DecodeStructSlice to avoid such discussions

Reply:
i would introduce generic codec.EncodeSlice[T scale.Encodeble](values []T) / codec.DecodeSlice[T scale.Decodable](buf []byte) ([]T, error) that will hide buffer and codec initialization in the codec module

Reply:
> i also not aware where we need to encode BlockIDs separately

@evt @dshulyak i am referring to here 
https://github.com/spacemeshos/go-spacemesh/blob/3a4cb1bc931eb6ca34bebb62f31beaa014e006a5/common/types/ballot_scale.go#L114

votes.support and votes.Against are both []types.BlockID, which is similar to []Transaction.ID. you used `scale.EncodeStructSlice` there. why not here also?

Reply:
@countvonzero this is generated by `scalegen` tool and perfectly works with a slice being a field in a struct. But not yet for a standalone slice.

Reply:
@evt i don't understand. those []TransactionID is not standalone. they also live in the struct InnerProposal.

Reply:
i guess you are saying you need them to be standalone for encoding in this sql function. ok. i got it.

Reply:
yeah, this is the only place where we run `codec.Encode` for a slice

Reply:
replaced  `TransactionIDs` with `EncodeSlice/DecodeSlice` as suggested by @dshulyak replaced  `TransactionIDs` with `EncodeSlice/DecodeSlice` as suggested by @dshulyak